include/libopencm3/stm32/g0/syscfg.h

```diff
@@ -0,0 +1,325 @@
+ * Guillaume Revaillot <g.revaillot@gmail.com>
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBOPENCM3_SYSCFG_H
+#define LIBOPENCM3_SYSCFG_H
+
+/**@{*/
+
+/**@defgroup syscfg_registers SYSCFG Registers
+￼  @{*/
+#define SYSCFG_CFGR1			MMIO32(SYSCFG_BASE + 0x00)
+#define SYSCFG_CFGR2			MMIO32(SYSCFG_BASE + 0x18)
+
+#define SYSCFG_ITLINE(line)		MMIO32(SYSCFG_BASE + 0x80 + (line)*4)
+/**@}*/
+
+/** @defgroup syscfg_cfgr1 CFGR1 SYSCFG configuration register 1
+@{*/
+
+
+/** SYSCFG_CFGR1_I2C_PA10_FMP FM+ enable for PA10 */
+#define SYSCFG_CFGR1_I2C_PA10_FMP		(1 << 23)
+
+/** SYSCFG_CFGR1_I2C_PA9_FMP FM+ enable for PA9 */
+#define SYSCFG_CFGR1_I2C_PA9_FMP		(1 << 22)
+
+/** SYSCFG_CFGR1_I2C2_FMP FM+ driving capability activation for I2C2 */
+#define SYSCFG_CFGR1_I2C2_FMP		(1 << 21)
+
+/** SYSCFG_CFGR1_I2C1_FMP FM+ driving capability activation for I2C1 */
+#define SYSCFG_CFGR1_I2C1_FMP		(1 << 20)
+
+/** SYSCFG_CFGR1_I2C_PB9_FMP FM+ enable for PB9 */
+#define SYSCFG_CFGR1_I2C_PB9_FMP		(1 << 19)
+
+/** SYSCFG_CFGR1_I2C_PB8_FMP FM+ enable for PB8 */
+#define SYSCFG_CFGR1_I2C_PB8_FMP		(1 << 18)
+
+/** SYSCFG_CFGR1_I2C_PB7_FMP FM+ enable for PB7 */
+#define SYSCFG_CFGR1_I2C_PB7_FMP		(1 << 17)
+
+/** SYSCFG_CFGR1_I2C_PB6_FMP FM+ enable for PB6 */
+#define SYSCFG_CFGR1_I2C_PB6_FMP		(1 << 16)
+
+/** SYSCFG_CFGR1_UCPD2_STROBE Strobe signal bit for UCPD2 */
+#define SYSCFG_CFGR1_UCPD2_STROBE		(1 << 10)
+
+/** SYSCFG_CFGR1_UCPD1_STROBE Strobe signal bit for UCPD1 */
+#define SYSCFG_CFGR1_UCPD1_STROBE		(1 << 9)
+
+/** SYSCFG_CFGR1_BOOSTEN I/O analog switch voltage booster enable */
+#define SYSCFG_CFGR1_BOOSTEN		(1 << 8)
+
+#define SYSCFG_CFGR1_IR_MOD_SHIFT			6
+#define SYSCFG_CFGR1_IR_MOD_MASK			0x03
+/** @defgroup syscfg_cfgr1_ir_mod IR MOD
+* @brief IR Modulation Envelope signal selection.
+@{*/
+#define SYSCFG_CFGR1_IR_MOD_TIM16			0
+#define SYSCFG_CFGR1_IR_MOD_USART1			1
+#define SYSCFG_CFGR1_IR_MOD_USART4			2
+/**@}*/
+
+/** SYSCFG_CFGR1_IR_POL IR output polarity selection */
+#define SYSCFG_CFGR1_IR_POL		(1 << 5)
+
+/** SYSCFG_CFGR1_PA12_RMP PA12 pin remapping */
+#define SYSCFG_CFGR1_PA12_RMP		(1 << 4)
+
+/** SYSCFG_CFGR1_PA11_RMP PA11 pin remapping */
+#define SYSCFG_CFGR1_PA11_RMP		(1 << 3)
```

Comment:
Very nice. These will come in handy for pin remapping on an 8 pin stm32g031j6

Reply:
Well, I need to recheck, but i think that on 8 pins chips, pins are actually bonded to multiple gpio.

Reply:
> but i think that on 8 pins chips, pins are actually bonded to multiple gpio.

Yes, this is correct. The remapping is an additional twist to the story.

I tested this PR yesterday and it worked. The default mapping for an stm32g031j6 is for PA11 and PA12 to be mapped to physical pins 5 and 6 respectively. By setting SYSCFG_CFGR1_PA11_RMP and SYSCFG_CFGR1_PA12_RMP PA9 and PA10 can be mapped to physical pins 5 and 6 instead of PA11 and PA12.